pytorch_lightning/trainer/trainer.py

```diff
@@ -765,7 +786,59 @@ def test(
             results = self.__test_using_best_weights(ckpt_path, test_dataloaders)
 
         self.teardown('test')
+        self.running_stage = RunningStage.UNDEFINED
+
+        return results
+
+    def predict(
+        self,
+        model: Optional[LightningModule] = None,
+        dataloaders: Optional[Union[DataLoader, List[DataLoader]]] = None,
+        verbose: bool = True,
+    ):
+        r"""
+
+        Separates from fit to make sure you never run on your predictions set until you want to.
+
+        Args:
+            model: The model to test.
+
+            dataloaders: Either a single
+                Pytorch Dataloader or a list of them, specifying inference samples.
+
+            verbose: If True, prints the test results
+
+        Returns:
+            The final test result dictionary. If no test_epoch_end is defined returns a list of dictionaries
+        """
+
+        # --------------------
+        # SETUP HOOK
+        # --------------------
+        self.running_stage = RunningStage.PREDICTING
+        self.verbose_test = verbose
+
+        if not dataloaders:
+            raise MisconfigurationException(
+                'You need to pass dataloaders to trainer.predict. '
+            )
+
```

Comment:
not sure if we should do more checking here:
```suggestion
if not isinstance(dataloaders, Dataloader) or (not isinstance(dataloaders, list) and all(isinstance(d, Dataloader) for d in dataloaders)):
    raise MisconfigurationException('``dataloaders`` argument either need to be a single pytorch dataloader or a list of dataloaders')
```